debian/hellanzb.1
.TH HELLANZB 1 "September 2006" "Version 0.10"
.SH NAME
hellanzb - a nzb downloader and post processor
.SH SYNOPSIS
.B
hellanzb
[options] [remote\-call] [remote\-call\-options]
.SH DESCRIPTION
Python application designed for *nix environments that retrieves nzb files and fully processes them. The goal being to make getting files from Usenet as hands-free as possible. Once fully installed, all thats required is moving an nzb file to the queue directory. The rest; fetching, par-checking, un-raring, etc. is taken care of by hellanzb.
.SH OPTIONS 
.B
hellanzb
will by default (no remote\-call specified) start its one and only queue daemon. Specifying a remote call will attempt to talk to that already running queue daemon via XML\-RPC.
.SS "remote-calls (via XML-RPC):"
.TP
cancel
Cancel the current download and move the current NZB to
Hellanzb.TEMP_DIR
.TP
clear
Clear the current nzb queue. Specify True as the second
argument to clear anything currently downloading as
well (like the cancel call)
.TP
continue
Continue downloading after being paused
.TP
dequeue nzbid
Remove the NZB with specified ID from the queue
.TP
down nzbid [shift]
Move the NZB with the specified ID down in the queue.
The optional second argument specifies the number of
spaces to shift by (Default: 1)
.TP
enqueue nzbfile
Add the specified NZB file to the end of the queue
.TP
enqueuenewzbin nzbid
Download the NZB with the specified NZB ID from
www.newzbin.com, and enqueue it
.TP
force nzbid
Force hellanzb to begin downloading the NZB with the
specified ID immediately, interrupting the current
download
.TP
last nzbid
Move the NZB with the specified ID to the end of the
queue
.TP
list [excludeids]
List the NZBs in the queue, along with their NZB IDs.
Specify True as the second argument to exclude the NZB
ID in the listing
.TP
maxrate [newrate]
Return the Hellanzb.MAX_RATE (maximum download rate)
value. Specify a second argument to change the value \fB\-\-\fR
a value of zero denotes no maximum rate
.TP
move nzbid index
Move the NZB with the specified ID to the specified
index in the queue
.TP
next nzbid
Move the NZB with the specified ID to the beginning of
the queue
.TP
pause
Pause downloading
.TP
process archivedir
Post process the specified directory. The \fB\-p\fR option is
preferable \fB\-\-\fR it will do this for you, or use the
current process if this XML\-RPC call fails
.TP
setrarpass nzbid pass Set the rarPassword for the NZB with the specified ID
.TP
shutdown
Shutdown hellanzb. Will quietly kill any post processing threads that may exist
.TP
status
Return hellanzb's current status text
.TP
up nzbid [shift]
Move the NZB with the specified ID up in the queue. The
optional second argument specifies the number of spaces
to shift by (Default: 1)
.SS "options:"
.TP
\fB\-\-version\fR
show program's version number and exit
.TP
\fB\-h\fR, \fB\-\-help\fR
show this help message and exit
.TP
\fB\-c\fR CONFIGFILE, \fB\-\-config\fR=\fICONFIGFILE\fR
specify the configuration file
.TP
\fB\-l\fR LOGFILE, \fB\-\-log\-file\fR=\fILOGFILE\fR
specify the log file (overwrites the Hellanzb.LOG_FILE
config file setting)
.TP
\fB\-d\fR DEBUGLOGFILE, \fB\-\-debug\-file\fR=\fIDEBUGLOGFILE\fR
specify the debug log file (turns on debugging
output/overwrites the Hellanzb.DEBUG_MODE config file
setting)
.TP
\fB\-D\fR, \fB\-\-daemon\fR
run hellanzb as a daemon process (fork and exit)
.TP
\fB\-p\fR POSTPROCESSDIR, \fB\-\-post\-process\-dir\fR=\fIPOSTPROCESSDIR\fR
post\-process the specified nzb archive dir either in
an already running hellanzb (via xmlrpc) if one is
available, otherwise in the current process. then exit
.TP
\fB\-P\fR RARPASSWORD, \fB\-\-rar\-password\fR=\fIRARPASSWORD\fR
when used with the \fB\-p\fR option, specifies the nzb
archive's rar password
.TP
\fB\-L\fR, \fB\-\-local\-post\-process\fR
when used with the \fB\-p\fR option, do the post processing
work in the current process (do not attempt to contact
an already running queue daemon)
.TP
\fB\-r\fR RPCSERVER, \fB\-\-rpc\-server\fR=\fIRPCSERVER\fR
specify the rpc server hostname (overwrites
Hellanzb.XMLRPC_SERVER config file setting)
.TP
\fB\-s\fR RPCPASSWORD, \fB\-\-rpc\-password\fR=\fIRPCPASSWORD\fR
specify the rpc server password (overwrites
Hellanzb.XMLRPC_PASSWORD config file setting)
.TP
\fB\-t\fR RPCPORT, \fB\-\-rpc\-port\fR=\fIRPCPORT\fR
specify the rpc server port (overwrites
Hellanzb.XMLRPC_PORT config file setting)
.SH FILES
.TP
.B $HOME/.hellanzb/hellanzb.conf 
User specific configuration file.
.TP
.B /etc/hellanzb.conf 
Default configuration file for all users (will be overrided by the file above).
.SH AUTHOR
This manual page was written by Adam Cecile <gandalf@le-vert.net> for the Debian system (but may be used by others).
Permission is granted to copy, distribute and/or modify this document under the terms of the GNU General Public License, Version 2 or any later version published by the Free Software Foundation
On Debian systems, the complete text of the GNU General Public License can be found in /usr/share/common-licenses/GPL.
.SH "SEE ALSO"
.B
Homepage: 
http://www.hellanzb.com/


debian/changelog
hellanzb (0.13-6.1) unstable; urgency=low

  * Non-maintainer upload.
  * Drop architecture restrictions from the Suggest field (closes: #581300).

 -- Jakub Wilk <jwilk@debian.org>  Fri, 16 Jul 2010 12:13:35 +0200

hellanzb (0.13-6) unstable; urgency=low

  * ACK previous NMUs, thanks a lot.
  * Merge patch from Eric Evans to fix compatibility with newer twisted
    releases, thanks (Closes: #573221).
  * Bump Standards-Version to 3.8.4.
  * Add missing ${misc:Depends} dependency.
  * Build depends on python-all instead of python-all-dev.

 -- Adam Cécile (Le_Vert) <gandalf@le-vert.net>  Fri, 12 Mar 2010 14:53:32 +0100

hellanzb (0.13-5.4) unstable; urgency=low

  * Non-maintainer upload.
  * Add debian/patches/006-Fix_embedded_nzb_postprocessing to fix
    post-processing in case of an embedded NZB file.
    Patch by Marcel de Vries (Closes: #556779).
  * Add debian/patches/100-python_26_fixes to fix Python 2.6
    deprecation warnings. Patch by Alessio Treglia (Closes: #555207).

 -- Severin Heiniger <severinheiniger@gmail.com>  Tue, 17 Nov 2009 20:14:52 +0100

hellanzb (0.13-5.3) unstable; urgency=low

  * Non-maintainer upload.
  * Bump Standards-Version to 3.8.3 (no changes needed).
  * Add new patch to fix crash when unrar isn't installed. (Closes: #554904)
  * Add README.sources to fix patch-system-but-no-source-readme lintian warning.
  * Minor cleanup in README.Debian and NEWS.Debian.

 -- Severin Heiniger <severinheiniger@gmail.com>  Sun, 08 Nov 2009 11:20:53 +0100

hellanzb (0.13-5.2) unstable; urgency=low

  * Non-maintainer upload.
  * Build-Depend on python-all-dev >= 2.5.4 for python.mk.

 -- Kumar Appaiah <akumar@debian.org>  Mon, 19 Oct 2009 18:08:56 -0500

hellanzb (0.13-5.1) unstable; urgency=low

  * Non-maintainer upload.
  * Fix "manipulates site-packages/ directly, failing with Python 2.6"
    Applied and uploaded Kumar's patch, thanks dude! (Closes: #547824)

 -- Bastian Venthur <venthur@debian.org>  Sat, 17 Oct 2009 16:37:22 +0200

hellanzb (0.13-5) unstable; urgency=low

  * Add python-openssl dependency (Closes: #520601).
  * Bump Standards-Version.

 -- Adam Cécile (Le_Vert) <gandalf@le-vert.net>  Tue, 31 Mar 2009 09:51:48 +0200

hellanzb (0.13-4) unstable; urgency=low

  * Fix mishandling of group values with extra whitespace (Closes: #495518).
  * Bump Standards-Version.

 -- Adam Cécile (Le_Vert) <gandalf@le-vert.net>  Wed, 11 Mar 2009 21:38:14 +0100

hellanzb (0.13-3) unstable; urgency=low

  * Merge 003-Fix_multiples_hosts patch, thanks to Benoit Boissinot
    (Closes: #452079).
  * Fix manpage 'SH NAME' section (bad whatis entry).
  * Bump Standards-Version to 3.7.3 (no changes needed).
  * Move homepage to new dpkg standard field.

 -- Adam Cécile (Le_Vert) <gandalf@le-vert.net>  Fri, 04 Jan 2008 23:39:34 +0100

hellanzb (0.13-2) unstable; urgency=low

  * Add "Recommends" on python-yenc (Closes: #418581).
  * Update supported archs for "unrar" suggest.
  * Add XS-Vcs fields.

 -- Adam Cécile (Le_Vert) <gandalf@le-vert.net>  Wed, 09 May 2007 09:15:22 +0200

hellanzb (0.13-1) unstable; urgency=low

  * New upstream release (Closes: #418580).
  * Uploading to unstable.

 -- Adam Cécile (Le_Vert) <gandalf@le-vert.net>  Tue, 10 Apr 2007 21:58:53 +0200

hellanzb (0.12-1) experimental; urgency=low

  * New upstream release (Closes: #412207).

 -- Adam Cécile (Le_Vert) <gandalf@le-vert.net>  Sat, 24 Feb 2007 20:37:04 +0100

hellanzb (0.11-1) experimental; urgency=low

  * New upstream release.
  * Update conffile search path patch.
  * Add a new patch to choose where the XMLRPC server will be binded,
    default 127.0.0.1 (Closes: #410349).
  * Do not ship a hellanzb.cnf anymore; patch and install the upstream one.
  * Attention: Configuration file has changed. See NEWS.Debian.

 -- Adam Cécile (Le_Vert) <gandalf@le-vert.net>  Sat, 10 Feb 2007 15:01:20 +0100

hellanzb (0.10-1) unstable; urgency=low

  * New upstream release.
    Update default conf file.
    Update conffile search path patch.
  * Add debian/watch.
  * Bump version in manpage, fix many typos.

 -- Adam Cécile (Le_Vert) <gandalf@le-vert.net>  Mon, 27 Nov 2006 01:21:30 +0100

hellanzb (0.9-3) unstable; urgency=low

  * debian/control: Drop XB-Python-Version.

 -- Adam Cécile (Le_Vert) <gandalf@le-vert.net>  Sun, 19 Nov 2006 03:26:50 +0100

hellanzb (0.9-2) unstable; urgency=low

  * Packaging update to fit the latest python policy draft.

 -- Adam Cécile (Le_Vert) <gandalf@le-vert.net>  Fri, 10 Nov 2006 01:55:36 +0100

hellanzb (0.9-1) unstable; urgency=low

  * Initial release (Closes: #386706).

 -- Adam Cécile (Le_Vert) <gandalf@le-vert.net>  Fri,  8 Sep 2006 21:12:36 +0200


hellanzb.py
#!/usr/bin/env python
"""

hellanzb - hella nzb

Command line interface to the Hellanzb library.

@see Hellanzb.Core.main()
@see Hellanzb.Core.parseArgs()

(c) Copyright 2005 Philip Jenvey, Ben Bangert
[See end of file]
"""
from Hellanzb.Core import main

__id__ = '$Id: hellanzb.py 665 2006-02-16 23:27:52Z pjenvey $'

if __name__ == '__main__':
    
    main()

"""
Copyright (c) 2005 Philip Jenvey <pjenvey@groovie.org>
                   Ben Bangert <bbangert@groovie.org>
All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions
are met:
1. Redistributions of source code must retain the above copyright
   notice, this list of conditions and the following disclaimer.
2. Redistributions in binary form must reproduce the above copyright
   notice, this list of conditions and the following disclaimer in the
   documentation and/or other materials provided with the distribution.
3. The name of the author or contributors may not be used to endorse or
   promote products derived from this software without specific prior
   written permission.

THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
SUCH DAMAGE.

$Id: hellanzb.py 665 2006-02-16 23:27:52Z pjenvey $
"""
